src/middlewared/middlewared/main.py

```diff
@@ -912,6 +912,9 @@ def event_subscribe(self, name, handler):
 
     def send_event(self, name, event_type, **kwargs):
         assert event_type in ('ADDED', 'CHANGED', 'REMOVED')
+
+        self.logger.trace(f'Sending event "{event_type}":{kwargs}')
```

Comment:
Passing formatted strings to logger is discouraged for many reasons; this should be `self.logger.trace('Sending event %r:%r', event_type, kwargs)`

Reply:
Can you share the reasons?